plankton/adapters/epics.py

```diff
@@ -23,10 +23,19 @@
 from datetime import datetime
 
 from . import Adapter, ForwardProperty
-from pcaspy import Driver, SimpleServer
 from six import iteritems
 
 from plankton.core.utils import seconds_since
+from plankton.core.exceptions import PlanktonException
+
+# pcaspy might not be available. To make EPICS-based adapters show up anyway,
+# dummy types are created in this case and the failure is postponed to
+# runtime, where a more appropriate PlanktonException can be raised.
+try:
+    from pcaspy import Driver, SimpleServer
+except ImportError:
+    Driver = type('Driver', (object,), {})
+    SimpleServer = type('SimpleServer', (object,), {})
```

Comment:
The downside of handling the import error in the client code is that every single client has to do it. If there are 10 devices with an EPICS interface, all 10 will have to `try`-`import`-`except`, this way it's only there once.

I think when using `--list-protocols` all adapters that are in the selected device module should be displayed, because otherwise users might not know that they exist. When they see one they want to use and try it, they will see the error describing what they have to do.

If the error is handled inside `import_adapter` the user will never be able to try and load the adapter and again never know it exists.